web/packages/teleport/src/JoinTokens/JoinTokens.tsx

```diff
@@ -0,0 +1,398 @@
+        <StyledLabel key={role}>{role}</StyledLabel>
+      ))}
+    </Cell>
+  );
+};
+
+const NameCell = ({ token }: { token: JoinToken }) => {
+  const { id, safeName, method } = token;
+  const [hovered, setHovered] = useState(false);
+  return (
+    <Cell
+      align="left"
+      style={{
+        minWidth: '320px',
+        fontFamily: 'monospace',
+        whiteSpace: 'nowrap',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+      }}
+      onMouseEnter={() => setHovered(true)}
+      onMouseLeave={() => setHovered(false)}
+    >
+      <Flex alignItems="center" gap={2}>
+        <Text
+          css={`
+            text-overflow: clip;
+            overflow-x: auto;
+          `}
+        >
+          {method !== 'token' ? id : safeName}
+        </Text>
+        {hovered && <CopyButton name={id} />}
+      </Flex>
+    </Cell>
+  );
+};
+
+const StyledLabel = styled(Label)`
+  height: 20px;
+  margin: 1px 0;
+  margin-right: ${props => props.theme.space[2]}px;
+  background-color: ${props => props.theme.colors.interactive.tonal.neutral[0]};
+  color: ${props => props.theme.colors.text.main};
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  line-height: 20px;
+`;
+
+function TokenDelete({
+  token,
+  onDelete,
+  onClose,
+  attempt,
+}: {
+  token: JoinToken;
+  onDelete: (token: string) => Promise<any>;
+  onClose: () => void;
+  attempt: Attempt<void>;
+}) {
+  return (
+    <Dialog
+      dialogCss={() => ({ maxWidth: '500px', width: '100%' })}
+      disableEscapeKeyDown={false}
+      onClose={close}
+      open={true}
+    >
+      <DialogHeader>
+        <DialogTitle>Delete Join Token?</DialogTitle>
+      </DialogHeader>
+      <DialogContent>
+        {attempt.status === 'error' && <Alert children={attempt.statusText} />}
+        <Text mb={4}>
+          You are about to delete join token
+          <Text bold as="span">
+            {` ${token.safeName}`}
+          </Text>
+          . This will not remove any resource that used this join token to join
+          the cluster. It will only prevent this token from being used to join
+          any more resources.
+        </Text>
+      </DialogContent>
+      <DialogFooter>
+        <ButtonWarning
+          mr="3"
+          disabled={attempt.status === 'processing'}
+          onClick={onDelete}
+        >
+          I understand, delete token
+        </ButtonWarning>
+        <ButtonSecondary onClick={onClose}>Cancel</ButtonSecondary>
+      </DialogFooter>
+    </Dialog>
+  );
+}
+
+const ActionCell = ({
+  onEdit,
+  onDelete,
+  token,
+}: {
+  onEdit(): void;
+  onDelete(): void;
+  token: JoinToken;
+}) => {
+  if (token.isStatic) {
+    return (
+      <Cell align="right">
+        <HoverTooltip
+          justifyContentProps={{ justifyContent: 'end' }}
+          tipContent="You cannot configure or delete static tokens via the web UI. Static tokens should be removed from your Teleport configuration file."
```

Comment:
I think we should change this to say "You can only configure or delete static tokens through tctl`, that way we convey they same thing but give the user the info they need to continue on

Reply:
Static tokens are in the actual teleport yaml config and not via tctl. They can only be removed by removing from the file

Reply:
Ah, I had no idea where they were managed - fair enough